File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/RecognitionMatrix.java
package org.firstinspires.ftc.teamcode.util;

import org.firstinspires.ftc.robotcore.external.matrices.OpenGLMatrix;
import org.firstinspires.ftc.robotcore.external.tfod.Recognition;

public class RecognitionMatrix {
    public Recognition recognition;
    public OpenGLMatrix position;
    public RecognitionMatrix(Recognition recognition) {
        this.recognition = recognition;
        float avgX = (recognition.getLeft() + recognition.getRight()) / 2;
        float avgY = (recognition.getTop() + recognition.getBottom()) / 2;
        this.position = OpenGLMatrix.translation(avgX, avgY, 0); // 2D matrix?
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/tfod/RecognitionMatrix.kt
package org.firstinspires.ftc.teamcode.util.tfod

import org.firstinspires.ftc.robotcore.external.matrices.OpenGLMatrix
import org.firstinspires.ftc.robotcore.external.tfod.Recognition

class RecognitionMatrix(val recognition: Recognition) {
    val position = OpenGLMatrix.translation(
        (recognition.left + recognition.right) / 2, // Average X
        (recognition.top + recognition.bottom) / 2, // Average Y
        0F // TF is in 2D
    )
}